pkg/gatewayserver/gatewayserver_test.go

```diff
@@ -664,9 +657,30 @@ func TestGatewayServer(t *testing.T) {
 					t.Run(tc.Name, func(t *testing.T) {
 						a := assertions.New(t)
 
+						testCtx := test.ContextWithT(test.Context(), t)
+						testCtx, cancel := context.WithTimeout(testCtx, 5*timeout)
+						defer cancel()
+
+						upEvents := map[string]events.Channel{}
+						for _, event := range []string{"gs.up.receive", "gs.down.tx.success", "gs.down.tx.fail", "gs.status.receive"} {
+							upEvents[event] = make(events.Channel, 5)
+						}
+						defer test.SetDefaultEventsPubSub(&test.MockEventPubSub{
+							PublishFunc: func(ev events.Event) {
+								switch name := ev.Name(); name {
+								case "gs.up.receive", "gs.down.tx.success", "gs.down.tx.fail", "gs.status.receive":
```

Comment:
I tried to refactor this into using the assertions in `test` package, but there are too many unhandled events published in the test script, so I adapted it in such a way.

Reply:
What would be the preferred way instead?

Reply:
Catch and assert all events published. (e.g. `gs.gateway.{dis,}connect`, `gs.up.drop`, etc.)